wordpress-mobile/WordPress-iOS
# My Site Dashboard: Sync posts more often and reload cards less

Part of #18308

## Description
This PR fixes an issue where posts weren't synced as often as before the refactor.
The issue was mentioned in https://github.com/wordpress-mobile/WordPress-iOS/pull/18351#issuecomment-1097115823
This issue was happening because the cards weren't being reloaded with every `viewDidAppear` event of the dashboard, due to the nature of diffable data sources.

The PR tackles this by:
- Add `DashboardPostsSyncManager` to be responsible for syncing all dashboard posts
- Use the new manager in `PostCardViewModel` instead of handling sync inside the view model
- Use the new manager to trigger syncing posts on `viewDidAppear` event of the dashboard
- Refactor posts parser, to reduce unnecessary reloading of dashboard cards, given that we now sync posts more often.

## Testing Instructions
1. Open a site with draft posts
2. Open wordpress.com from a browser
3. Trash one of the drafts from the browser
4. Wait till the trashing action is successful
5. On the app, navigate to the site menu and back
6. Make sure your change has been reflected.

## Regression Notes
1. Potential unintended areas of impact
    - Posts dashboard cards

4. What I did to test those areas of impact (or what existing automated tests I relied on)
    - Tested manually and added relevant unit tests

5. What automated tests I added (or what prevented me from doing so)
    - `DashboardPostsSyncManagerTests`
    - `testTrashingAPostWillUpdateItsRevisionStatusAfterSyncProperty`

PR submission checklist:
